### DISCUSSION TITLE
For loop to iterate through maps

### DISCUSSION THREAD

**Discussion by carbonvibes** [bpftrace/bpftrace#3041] [Q&A] [answered]
is it possible to write a for loop that iterates through a map and print values stored??

**Comment by jordalgo** [comment]
Soon!! https://github.com/bpftrace/bpftrace/pull/3003 😄

**Reply by carbonvibes** [reply]
if possible can you please let me know approximately how long would it take for [ #3003 ](https://github.com/bpftrace/bpftrace/pull/3003) to be implemented ?  i kinda need that for my project 😅

**Reply by jordalgo** [reply, marked as answer]
We're really hoping to land it within the next week or two because it's also blocking other projects. I'll try to remember to notify on this thread when it gets committed.
